### PR TITLE
[DOCS] Clarify `orientation` usage for WKT and GeoJSON polygons

### LINKED ISSUE #84009: Cannot capture New Zealand using Orientation
### Elasticsearch Version

7.16.2

### Installed Plugins

_No response_

### Java Version

openjdk version "1.8.0_292"

### OS Version

Darwin Kernel Version 20.6.0

### Problem Description

I'm using ES 7.16.2 and trying to capture New Zealand and having a horrible time of it.

This query does not capture NZ no matter the combinations of orientation values that I use.
```
POST /geo-test/_doc
{
  "location" : {
    "type" : "polygon",
    "coordinates" : [
      [
                [-175.1235077, -28.8773225],
                [165.7437641, -28.8773225],
                [165.7437641, -52.7224663],
                [-175.1235077, -52.7224663],
                [-175.1235077, -28.8773225]
       ]     
    ]
  }
}
```

The only way I can get it to work is by translating the negative longitudes by +360.

```
POST /geo-test/_doc
{
  "location" : {
    "type" : "polygon",
    "coordinates" : [
      [
                [185.1235077, -28.8773225],
                [165.7437641, -28.8773225],
                [165.7437641, -52.7224663],
                [185.1235077, -52.7224663],
                [185.1235077, -28.8773225]
       ]     
    ]
  }
}
```
Is that the intended method?

Note: I am viewing these polygons in the Kibana Map viewer. I suppose the issue could be there or I just really don't understand what Orientation is supposed to be doing here.

Thanks!

### Steps to Reproduce

Create index: 

```
PUT /geo-test
{
  "mappings":{
    "properties": {
      "location": {
        "type": "geo_shape"
      }
    }
  }
}
```

Use the queries listed above

View in Kibana Map viewer

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by jrodewig**
Clarifies that the `orientation` mapping parameter only applies to WKT polygons. GeoJSON polygons use a default orientation of `RIGHT`, regardless of the mapping parameter.

Also notes that the document-level `orientation` parameter overrides the default orientation for both WKT and GeoJSON polygons.

Closes https://github.com/elastic/elasticsearch/issues/84009.

### Preview
https://elasticsearch_84025.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/geo-shape.html#geo-polygon

### CODE REVIEW

**Comment by jrodewig** [comment]
Thanks for the clarification @iverase. Rather than re-add the previous note as-is, I reworked the current polygon orientation docs to include the relevant content. Let me know what you think.

File: docs/reference/mapping/types/geo-shape.asciidoc [hunk 1]
```diff
@@ -257,15 +257,22 @@ POST /example/_doc
...
-For example, the following indexing request specifies a document-level
-`orientation` of `LEFT`.
+GeoJSON polygons use a default orientation of `RIGHT`, regardless of
+`orientation` mapping parameter's value. This is because the
+https://tools.ietf.org/html/rfc7946#section-3.1.6[GeoJSON specification]
+mandates that an outer polygon use a counterclockwise orientation and interior
+shapes use a clockwise orientation.
+
+You can override the default orientation for WKT and GeoJSON polygons using the
+document-level `orientation` parameter. For example, the following indexing
```
Comment (iverase) [inline docs/reference/mapping/types/geo-shape.asciidoc:274]: I don't think this is true for WKT as you cannot add a document level orientation, that's the reason you need to do it in the mapping. 

For GeoJson it os done as shown in the example

**Review by iverase — CHANGES_REQUESTED** [review]
We need to do just a small correction.

While we are here, should we update the Geojson examples so the types are properly capitalise? 

Check it here: https://en.wikipedia.org/wiki/GeoJSON

**Review by iverase — APPROVED** [review]
Thanks for your patience @jrodewig!

**Comment by jrodewig** [comment]
Thanks @iverase!

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.1|[](https://github.com/elastic/elasticsearch/pull/84130)|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/84131)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/84132)|
|✅|7.16|[](https://github.com/elastic/elasticsearch/pull/84133)|
|✅|7.15|[](https://github.com/elastic/elasticsearch/pull/84134)|
|✅|7.14|[](https://github.com/elastic/elasticsearch/pull/84135)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**